token/transfer-hook/cli/src/meta.rs

```diff
@@ -0,0 +1,318 @@
+}
+impl ExtraAccountMetaConfig {
+    fn discriminator_and_address(&self) -> (u8, [u8; 32]) {
+        if self.pubkey.is_some() && self.seeds.is_some() {
+            panic!("Only one of `pubkey` or `seeds` must be present");
+        }
+        if let Some(pubkey_string) = &self.pubkey {
+            (0, Pubkey::from_str(pubkey_string).unwrap().to_bytes())
+        } else if let Some(seeds) = &self.seeds {
+            (1, Seed::pack_into_address_config(seeds).unwrap())
+        } else {
+            panic!("Either `pubkey` or `seeds` must be present");
+        }
+    }
+
+    fn bools(&self) -> (/* is_signer */ bool, /* is_writable */ bool) {
+        if self.role.is_none() && self.is_signer.is_none() && self.is_writable.is_none() {
+            panic!("Either `role` or `is_signer`/`is_writable` must be present");
+        }
+        if let Some(role) = self.role {
+            if self.is_signer.is_some() || self.is_writable.is_some() {
+                panic!("`role` and `is_signer`/`is_writable` are mutually exclusive");
+            }
+            role.bools()
+        } else {
+            if self.is_signer.is_none() || self.is_writable.is_none() {
+                panic!("`is_signer` and `is_writable` must be present when `role` is not present");
+            }
+            (self.is_signer.unwrap(), self.is_writable.unwrap())
+        }
+    }
+}
+impl From<&ExtraAccountMetaConfig> for ExtraAccountMeta {
+    fn from(config: &ExtraAccountMetaConfig) -> Self {
+        let (discriminator, address_config) = config.discriminator_and_address();
+        let (is_signer, is_writable) = config.bools();
+        ExtraAccountMeta {
+            discriminator,
+            address_config,
+            is_signer: is_signer.into(),
+            is_writable: is_writable.into(),
+        }
+    }
+}
+
+pub fn parse_transfer_hook_account_arg(arg: &str) -> Result<Vec<ExtraAccountMeta>, String> {
+    match arg.split(':').collect::<Vec<_>>().as_slice() {
+        [address, role] => {
+            let pubkey = Pubkey::from_str(address).map_err(|e| format!("{e}"))?;
+            let role = AccountMetaRole::from_str(role).map_err(|e| format!("{e}"))?;
+            let (is_signer, is_writable) = role.bools();
+            let meta = ExtraAccountMeta::new_with_pubkey(&pubkey, is_signer, is_writable)
+                .map_err(|e| format!("{e}"))?;
+            Ok(vec![meta])
+        }
+        _ => {
+            let path = Path::new(arg);
+            let parse_fn = match path.extension().and_then(|s| s.to_str()) {
+                Some("json") => |v: &str| {
+                    serde_json::from_str::<ExtraAccountMetaConfigFile>(v)
+                        .map_err(|e| format!("Unable to parse file: {e}"))
+                },
+                Some("yaml") => |v: &str| {
+                    serde_yaml::from_str::<ExtraAccountMetaConfigFile>(v)
+                        .map_err(|e| format!("Unable to parse file: {e}"))
+                },
```

Comment:
nit-for-safety: let's also match on `yml`
```suggestion
                Some("yaml") | Some("yml") => |v: &str| {
                    serde_yaml::from_str::<ExtraAccountMetaConfigFile>(v)
                        .map_err(|e| format!("Unable to parse file: {e}"))
                },
```